$NetBSD: patch-ae,v 1.1 2009/08/01 16:22:26 dholland Exp $

Hack around silly behavior that unlocks every newly created pthread mutex.

--- src/db/dbcontrol.c.orig	2009-08-01 12:21:37.000000000 -0400
+++ src/db/dbcontrol.c	2009-08-01 12:22:18.000000000 -0400
@@ -499,6 +499,6 @@ BOOLEAN ensurevalidmutex(void **mutex, B
 			return(TRUE);
 		}
 	}
-	emutexunlock(*mutex);
+	emutexensureunlocked(*mutex);
 	return(FALSE);
 }
